src/bootstrap/cc_detect.rs

```diff
@@ -147,32 +155,35 @@ fn set_compiler(cfg: &mut cc::Build,
             let c = cfg.get_compiler();
             let gnu_compiler = compiler.gcc();
             if !c.path().ends_with(gnu_compiler) {
-                return
+                return false;
             }
 
             let output = output(c.to_command().arg("--version"));
             let i = match output.find(" 4.") {
                 Some(i) => i,
-                None => return,
+                None => return false,
             };
             match output[i + 3..].chars().next().unwrap() {
                 '0' ..= '6' => {}
-                _ => return,
+                _ => return false,
             }
             let alternative = format!("e{}", gnu_compiler);
             if Command::new(&alternative).output().is_ok() {
                 cfg.compiler(alternative);
+                return true;
```

Comment:
I think that the `return true` here and eslewehre in this function may not be what we want? This all seems largely about auto-inference of what gcc should be used, but it doesn't imply that it was configured this way in particular. Notably I think that we should only actually probe for a C++ compiler if it's explicitly configured.

Note though that the C++ compiler can be configured not only via `config.toml` but also the standard `CXX_...` variables that `cc` supports. In that sense I don't think the return value from this function is correct.

Reply:
This was to behave exactly as before for hosts, but I'm 100% fine with dropping that.

We at least need to set it to the default 'c++' for the build hosts though otherwise the build fails on CI (or we need to explicitely configure cxx in the ci)

Reply:
For hosts, yes, but this logic seems like if you configure a target for one of these targets and you're missing the inferred C++ compiler it will fail, right?